### ISSUE TITLE
Bug: Some gLTF files do not have target names(aiAnimMesh name) properly loaded.

### ISSUE THREAD

**Issue by Futuremappermydud** [assimp/assimp#5354]
**Describe the bug**

When loading a binary glTF2 file and iterating over each animMesh in each mesh, some models have empty names for the animMesh. 

Edit: Cause in comment below

**To Reproduce**
```
#include <assimp/Importer.hpp>
#include <assimp/postprocess.h>
#include <assimp/scene.h>
#include <iostream>

int main()
{
    Assimp::Importer importer;

    const aiScene* scene = importer.ReadFile("C:/Users/User/OneDrive/Desktop/AliciaSolidFIX.gltf", aiProcess_Triangulate | aiProcess_LimitBoneWeights);

    for (size_t i = 0; i < scene->mNumMeshes; i++)
    {
        auto mesh = scene->mMeshes[i];
        for (size_t k = 0; k < mesh->mNumAnimMeshes; k++)
        {
            auto animMesh = mesh->mAnimMeshes[k];
            std::cout << animMesh->mName.C_Str() << std::endl;
        }
    }

    return 0;
}
```
1. Run this code with the path directed to a broken model
2. See empty lines(names are empty/whitespace)

**Expected behavior**
Each line should have the proper name of the animMesh, as seen inside the gltf File

**Platform (please complete the following information):**
 - OS: Android

**Additional context**
Models to test with https://drive.google.com/file/d/18xKZ-ZsITdNvlD1IAdobExMYeeY0EOzS/view?usp=drive_link

**Comment by Futuremappermydud** [comment]
https://pastebin.com/30SDn05U This paste shows the output of the following code (Has more logging)

```
#include <assimp/Importer.hpp>
#include <assimp/postprocess.h>
#include <assimp/scene.h>
#include <iostream>

int main()
{
    Assimp::Importer importer;

    const aiScene* scene = importer.ReadFile("C:/Users/Owens/OneDrive/Desktop/AliciaSolidFIX.gltf", aiProcess_Triangulate | aiProcess_LimitBoneWeights);

    for (size_t i = 0; i < scene->mNumMeshes; i++)
    {
        auto mesh = scene->mMeshes[i];
        std::cout << "mesh " << i << ": " << mesh->mName.C_Str() << std::endl;
        for (size_t k = 0; k < mesh->mNumAnimMeshes; k++)
        {
            auto animMesh = mesh->mAnimMeshes[k];
            std::cout << "blendshape " << k << ":" << animMesh->mName.C_Str() << ":size: " << animMesh->mName.length << std::endl;
        }
    }

    return 0;
}
```

**Comment by Futuremappermydud** [comment]
Doing some more research into assimp and the jsons of working and nonworking meshes I found the issue.

Broken model: https://pastebin.com/9zasPYfD
Working model: https://pastebin.com/XChyNrbs
Broken LOCs: https://github.com/assimp/assimp/blob/master/code/AssetLib/glTF2/glTF2Asset.inl#L1569-L1580

The problem seems to be that the broken model has the target names inside of each primitive, not inside the root of the mesh like the working model.

**Comment by Futuremappermydud** [comment]
Should be fixed in #5356

**Comment by kimkulling** [comment]
Thanks for the PR, I will review it.
